common/unsafe/src/main/java/org/apache/spark/unsafe/DoubleAccessFunctor.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.unsafe;
+
+import sun.misc.Unsafe;
+
+interface DoubleAccessFunctor {
```

Comment:
Interesting @kiszk I had not encountered this before. I would have used an interface for simplicity, but I understand this is an argument about performance in hot code. Design-wise it's virtually identical. According to http://stackoverflow.com/questions/21608240/performance-difference-between-passing-interface-and-class-reloaded this particular case might be just as fast. I think we can leave it unless someone has a strong feeling otherwise.